packages/utils/src/schema/retrieveSchema.ts

```diff
@@ -192,6 +195,39 @@ export function resolveReference<T = any, S extends StrictRJSFSchema = RJSFSchem
   );
 }
 
+/** Resolves all references within a schema's properties and array items.
+ *
+ * @param schema - The schema for which resolving all references is desired
+ * @param rootSchema - The root schema that will be forwarded to all the APIs
+ * @returns - given schema will all references resolved
+ */
+export function resolveAllReferences<S extends StrictRJSFSchema = RJSFSchema>(schema: S, rootSchema: S): S {
```

Comment:
@heath-freenome I was debating making this util specifically for oneOf/anyOf options. This is already assuming that the schema went through `resolveSchema` and has been evaluated with `formData`. Its just getting the defined $ref and updating the schema.

This might not solve the core issue of oneOf/anyOf options not fully being resolved in `resolveSchema` but it does solve the existing issue with minimum impact on the whole resolve workflow.